File: easy/88_vigenere_cipher/cipher.hs
import Data.Char

encrypt :: String -> String -> String
encrypt p k = zipWith forward p (cycle k)

charValue :: Char -> Int
charValue a = (ord . toLower) a - 97

forward :: Char -> Char -> Char
forward a b = chr $ ((charValue a + charValue b) `mod` 26) + 97

main = do
   plaintext <- getLine
   key <- getLine
   putStrLn $ encrypt plaintext key

File: easy/88_vigenere_cipher/cipher.py
from pprint import pprint
import itertools

def encode(plaintext, key):

   # zip lists into tuple, stretching smaller list to match upper
   zipped = [(i,j) for i, j in itertools.izip(plaintext, itertools.cycle(key))]
   
   # add the two letter values (from 0 to 26), and conver back to char
   enc = lambda x, y: chr( ((x + y) % 26) + 97 )

   # run enc lambda for each letter value, join back into string
   return ''.join([enc(ord(x) - 97, ord(y) - 97) for x, y in zipped])
      

if __name__ == '__main__':
   plaintext = [x.lower() for x in 'THECAKEISALIE']
   key =       [x.lower() for x in 'GLADOS']

   pprint(encode(plaintext, key))
   